fe/src/main/java/org/apache/doris/qe/Coordinator.java

```diff
@@ -1469,8 +1470,41 @@ public boolean isBackendStateHealthy() {
             try {
                 return BackendServiceProxy.getInstance().execPlanFragmentAsync(brpcAddress, rpcParams);
             } catch (RpcException e) {
-                SimpleScheduler.addToBlacklist(backend.getId());
-                throw e;
+                // Do not throw exception here.
+                // return a completed future, the caller will check the result and do the cancel process.
+                PExecPlanFragmentResult result = new PExecPlanFragmentResult();
+                PStatus pStatus = new PStatus();
+                pStatus.error_msgs.add(e.getMessage());
+                // use THRIFT_RPC_ERROR so that this BE will be added to the black list later.
```

Comment:
```suggestion
                // use THRIFT_RPC_ERROR so that this BE will be added to the black list later.
```
// use THRIFT_RPC_ERROR so that this BE will be added to the blacklist later.